testsuite/forge/src/backend/k8s/swarm.rs

```diff
@@ -460,19 +460,31 @@ pub fn k8s_wait_nodes_strategy() -> impl Iterator<Item = Duration> {
     fixed_retry_strategy(10 * 1000, 120)
 }
 
+pub fn k8s_wait_indexer_strategy() -> impl Iterator<Item = Duration> {
+    // retry every 10 seconds for 20 minutes
+    fixed_retry_strategy(10 * 1000, 120)
+}
+
 async fn list_stateful_sets(client: K8sClient, kube_namespace: &str) -> Result<Vec<StatefulSet>> {
     let stateful_set_api: Api<StatefulSet> = Api::namespaced(client, kube_namespace);
     let lp = ListParams::default();
     let stateful_sets = stateful_set_api.list(&lp).await?.items;
     Ok(stateful_sets)
 }
 
-fn stateful_set_name_matches(sts: &StatefulSet, suffix: &str) -> bool {
-    if let Some(s) = sts.metadata.name.as_ref() {
-        s.contains(suffix)
-    } else {
-        false
+/// Check if the stateful set labels match the given labels
+fn stateful_set_labels_matches(sts: &StatefulSet, labels: &BTreeMap<String, String>) -> bool {
```

Comment:
We probably won't hit this case since usually we don't autogenerate labels the same way we do with names, since we generally stick with labels for `app`/`instance`, etc. Doesn't hurt to add a check I suppose, though